test/Conversion/tritongpu_to_llvm.mlir

```diff
@@ -1455,3 +1455,20 @@ module attributes {"triton_gpu.compute-capability" = 80 : i32, "triton_gpu.num-c
     tt.return
   }
 }
+
+// -----
+
+// CHECK-LABEL: dont_divide_0
+// CHECK: %[[C0:.*]] = llvm.mlir.constant(0 : i32) : i32
+// CHECK-NOT: llvm.urem %{{.*}}, %[[C0]]
+// CHECK: %[[C1:.*]] = llvm.mlir.constant(0 : i32) : i32
+// CHECK-NOT: llvm.urem %{{.*}}, %[[C1]]
```

Comment:
There are multiple `llvm.mlir.constant(0 : i32)`, and the second one seems to  be the problematic one (fails without this PR), but I'm not sure if there's a way to check all of them.

Reply:
Don't think there's a way to do that in LLVM. Is this test supposed to produce any `llvm.urem` instructions? If not, maybe we can just `CHECK-NOT: llvm.urem`?

Reply:
Yes it is and multiple of them.  This change actually just changes one of the `llvm.urem`s that it generates from a `llvm.urem x, 0` to `llvm.urem x, 1`.

Both before & after this change, this test produces multiple `constant(0` and `llvm.urem`; I'm really just trying to check that we are not adding an `llvm.urem x, 0`

Reply:
CSE pass does work on LLVM dialect. It would merge constant ops

Reply:
Thanks; added.